File: MainModule/src/main/java/com/weidi/mainmodule/MainActivity.java
package com.weidi.mainmodule;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.alibaba.android.arouter.facade.Postcard;
import com.alibaba.android.arouter.facade.callback.NavigationCallback;
import com.alibaba.android.arouter.launcher.ARouter;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.bt_1).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ARouter.getInstance().build("/path/moduleA").navigation();
            }
        });


        findViewById(R.id.bt_2).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ARouter.getInstance().build("/path/activity2")
                        .withString("from","MainModule").navigation();
            }
        });

        findViewById(R.id.bt_3).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ARouter.getInstance().build("/moduleB/activity1")
                        .navigation(
//                                getBaseContext(), new NavCallback() {
//                            @Override
//                            public void onArrival(Postcard postcard) {
//                                Log.e("===","===arrive");
//                            }
//
//                            @Override
//                            public void onInterrupt(Postcard postcard) {
//                                super.onInterrupt(postcard);
//                                Log.e("===","===BEILANJIE");
//
//                            }
//                        }
                        );
            }
        });

        findViewById(R.id.bt_4).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ARouter.getInstance().build("/modulec/main").navigation(getBaseContext(), new NavigationCallback() {
                    @Override
                    public void onFound(Postcard postcard) {
                        Log.e("=========found","modulec postcard = "+postcard.toString());
                    }

                    @Override
                    public void onLost(Postcard postcard) {

                    }

                    @Override
                    public void onArrival(Postcard postcard) {
                        Log.e("=========","modulec postcard = "+postcard.toString());
                    }

                    @Override
                    public void onInterrupt(Postcard postcard) {

                    }
                });
            }
        });

        findViewById(R.id.bt_5).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ARouter.getInstance().build("/moduled/main").navigation();
            }
        });
    }
}

File: ModuleC/src/main/java/com/weidi/modulec/MainActivity.kt
package com.weidi.modulec

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.alibaba.android.arouter.facade.annotation.Route

@Route(path = "/modulec/main")
class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        return setContentView(R.layout.activity_main1)
        Log.e("===========modulec====","=====create")
    }

    override fun onStart() {
        super.onStart()
        Log.e("===========modulec====","=====start")
    }

    override fun onStop() {
        super.onStop()
        Log.e("===========modulec====","=====stop")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.e("===========modulec====","=====destroy")
    }

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        Log.e("===========modulec====","=====attached")
    }

    override fun onDetachedFromWindow() {
        super.onDetachedFromWindow()
        Log.e("===========modulec====","=====detached")
    }
}
